src/tests/ReactUnusedPropsAndStateRuleTests.ts

```diff
@@ -614,4 +614,415 @@ describe('reactUnusedPropsAndStateRule', (): void => {
+                        readonly somethingElse: string,
+                    }> = (p) => (
+                        <span>{p.text}</span>
+                    )
+                `;
+
+                TestHelper.assertViolations(ruleName, script, [
+                    {
+                        failure: 'Unused React property defined in interface: somethingElse',
+                        name: Utils.absolutePath('file.tsx'),
+                        ruleName: 'react-unused-props-and-state',
+                        startPosition: { character: 25, line: 6 }
+                    }
+                ]);
+            });
+
+            it('finds missing prop on example from desktop/desktop', () => {
+                const script = `
+                import React = require('react');
+
+                interface Props {
+                  readonly text: string
+                  readonly highlight: ReadonlyArray<number>
+                  readonly somethingElse: string
+                }
+
+                export const HighlightText: React.SFC<Props> = (props) => (
+                  <span>
+                    {
+                      props.text
+                        .split('')
+                        .map((ch, i): [string, boolean] => [ch, props.highlight.includes(i)])
+                        .concat([['', false]])
+                        .reduce(
+                          (state, [ch, matched], i, arr) => {
+                            if (matched === state.matched && i < arr.length - 1) {
+                              state.str += ch
+                            } else {
+                              const Component = state.matched ? 'mark' : 'span'
+                              state.result.push(<Component key={i}>{state.str}</Component>)
+                              state.str = ch
+                              state.matched = matched
+                            }
+                            return state
+                          },
+                          {
+                            matched: false,
+                            str: '',
+                            result: new Array<React.ReactElement<any>>(),
+                          }
+                        ).result
+                    }
+                  </span>
+                )`;
+
+                TestHelper.assertViolations(ruleName, script, [
+                    {
+                        failure: 'Unused React property defined in interface: somethingElse',
+                        name: Utils.absolutePath('file.tsx'),
+                        ruleName: 'react-unused-props-and-state',
+                        startPosition: { character: 19, line: 7 }
+                    }
+                ]);
+            });
+
+            // TODO: this test is a thorny one to resolve, and needs some further thought
+            //       about how to address this with SFCs
+            it.skip('finds missing prop when ObjectBindingExpression used', () => {
```

Comment:
Putting this back to `WIP` to investigate how I might solve this test case (I could diff the identifiers inside the `ObjectBindingExpression` against `propNames` and if there are values missing, this could immediately report an error).

Reply:
Should be fixed in b1e907a